src/model/MutableOrganizationDataSource.ts

```diff
@@ -0,0 +1,158 @@
+    const session = await this.organizationModel.startSession()
+    let ret: OrganizationType
+    // withTransaction() doesn't return the callback result
+    // see https://jira.mongodb.org/browse/NODE-2014
+    await session.withTransaction(
+      async (session) => {
+        ret = await this._addOrganization(session, user, displayName, orgType)
+        return ret
+      })
+    // @ts-expect-error
+    return ret
+  }
+
+  async _addOrganization (session, user: MUUID, displayName: string, orgType: OrgType): Promise<any> {
+    const change = await changelogDataSource.create(session, user, OperationType.addOrganization)
+    const newChangeMeta: ChangeRecordMetadataType = {
+      user: user,
+      historyId: change._id,
+      operation: OperationType.addOrganization,
+      seq: 0
+    }
+
+    const newOrg = newOrganizationHelper(displayName, orgType)
+    newOrg.createdBy = user
+    newOrg._change = produce(newChangeMeta, draft => {
+      draft.seq = 1
+    })
+
+    const rs1 = await this.organizationModel.insertMany(newOrg, { session })
+    return rs1[0].toObject()
+  }
+
+  /**
+   * Update one or more organization fields
+   *
+   * @param user User requesting the update
+   * @param orgId Uuid of the organization to be updated
+   * @param document New fields
+   * @returns Newly updated organization
+   */
+  async updateOrganization (user: MUUID, orgId: MUUID, document: OrganizationEditableFieldsType): Promise<OrganizationType | null> {
+    const _updateOrganization = async (session: ClientSession, user: MUUID, orgId: MUUID, document: OrganizationEditableFieldsType): Promise<any> => {
+      const filter = {
+        orgId: orgId,
+        deleting: { $ne: null }
+      }
+
+      const org = await this.organizationModel.findOne(filter).session(session)
+
+      if (org == null) {
+        throw new Error('Organization update error. Reason: Organization not found.')
+      }
+
+      const { associatedAreaIds, excludedAreaIds, displayName, website, email, donationLink, instagramLink, description } = document
+
+      if (associatedAreaIds.length > 0) {
+        const missingAreaIds = await findNonexistantAreas(associatedAreaIds)
+        if (missingAreaIds.length > 0) throw new Error(`Organization update error. Reason: Associated areas not found: ${missingAreaIds.map(m => m.toUUID().toString()).toString()}`)
+        org.set({ associatedAreaIds: associatedAreaIds })
+      }
+      if (excludedAreaIds.length > 0) {
+        const missingAreaIds = await findNonexistantAreas(excludedAreaIds)
+        if (missingAreaIds.length > 0) throw new Error(`Organization update error. Reason: Excluded areas not found: ${missingAreaIds.map(m => m.toUUID().toString()).toString()}`)
+        org.set({ excludedAreaIds: excludedAreaIds })
+      }
+      if (displayName != null) { org.set({ displayName: sanitizeStrict(displayName) }) }
+      if (website != null) { org.set({ 'content.website': sanitizeStrict(website) }) }
```

Comment:
We could do alot more to validate that this is a well-formed url. Same for emails below. But not sure how opinionated to be here.